### DISCUSSION TITLE
Interfacing to 8266-01S doesn't work

### DISCUSSION THREAD

**Discussion by HaikoKurt** [micropython/micropython#11808] [ESP8266] [answered]
Hello,

I'm trying to install Micopython on an ESP8266-S01. Deleting and programming the firmware works. When I turn on the board, the blue LED flashes quickly. When I switch on the try interface with 115200 baud I get weird character output on screen. Maybe I have the wrong baud rate?

I tried two different firmware versions: 1.20.0 and 1.14

Thanks for help

Haiko

**Comment by cwalther** [comment]
Try 74880 baud to see the boot messages. However, once MicroPython takes over, it switches to 115200 so you should see something there. If it’s constantly rebooting: the most common cause for that is a wrong flash mode. Try different values for the `-fm` argument of `esptool.py`.

**Reply by HaikoKurt** [reply]
Thank you for answering. 

i tried all five flash modes {keep,qio,qout,dio,dout}. Always with the same result. I flashed with the following command line:

esptool --port /dev/usbserial-1111120 write_flash -fm dout 0x00000 micropython/esp8266-1m-20230426-v1.20.0.bin

After that I tried to connect to the ESP with:

screen /dev/tty.usbserial-1111120 74880

Am I using wrong command parameters?

Thanks for help

Haiko

**Reply by cwalther** [reply]
I don’t think `screen` supports the non-standard baud rate 74880 on macOS (it doesn’t seem to here). Try
```
miniterm.py --raw /dev/tty.usbserial-1111120 74880
```
`miniterm.py` comes with pySerial which you should already have as a dependency of `esptool.py`.

If you are sure you are receiving at 74880 and still get garbled boot messages, it could be that your module [has a mismatched crystal](https://forum.micropython.org/viewtopic.php?p=58776#p58776).

**Reply by HaikoKurt** [reply]
Thanks Christian,

Miniterm helps! Why such a crude baud rate is used is not to be understood. Now I get a readable output:

 ets Jan  8 2013,rst cause:2, boot mode:(3,6)

load 0x40100000, len 30756, room 16 
tail 4
chksum 0xfd
load 0x3ffe8000, len 1016, room 4 
tail 4
chksum 0x52
load 0x3ffe8400, len 1080, room 4 
tail 4
chksum 0xa1
csum 0xa1
rf_cal[0] !=0x05,is 0xF0

i guess the micropython doesn't start properly. Otherwise the baudrate would change to 115200. What's the problem?

Thanks Haiko

**Reply by cwalther** [reply]
I’m afraid that’s where my expertise ends. The last line seems to suggest something is wrong with radio calibration data? Google spits out this: https://arduino.stackexchange.com/questions/33590/endless-loop-on-boot-after-reflashing-esp-12e-with-at-firmware

**Comment by robert-hh** [comment]
If you ate sure that the device has 1MB flash, you could also manually specify the flash size with the -fs option. Sometimes the auto detection does not work.

**Comment by scruss** [comment, marked as answer]
[this](https://forum.micropython.org/viewtopic.php?f=2&t=12718&p=70720&hilit=esp01#p70720) forum post suggests the correct esptool command line for you might be:

    esptool.py --port /dev/usbserial-1111120 --baud 460800 write_flash --flash_mode dout --flash_size=detect 0 micropython/esp8266-1m-20230426-v1.20.0.bin

**Reply by HaikoKurt** [reply]
Perfect! Now it works. Thank You
